src/platform/plugins/shared/data/server/deprecations/search_sessions.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type {
+  CoreSetup,
+  DeprecationsDetails,
+  GetDeprecationsContext,
+  RegisterDeprecationsConfig,
+  SavedObjectsFindResult,
+} from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+import type { DeprecationDetailsMessage } from '@kbn/core-deprecations-common';
+import { SEARCH_SESSION_TYPE, SearchSessionSavedObjectAttributes } from '../../common';
+
+type SearchSessionAttributes = Pick<
+  SearchSessionSavedObjectAttributes,
+  'name' | 'username' | 'expires'
+>;
+
+export const createSearchSessionsDeprecationsConfig: (
+  core: CoreSetup
+) => RegisterDeprecationsConfig = (core: CoreSetup) => ({
+  getDeprecations: async (context: GetDeprecationsContext): Promise<DeprecationsDetails[]> => {
+    const [coreStart] = await core.getStartServices();
+    const savedObjectsClient = coreStart.savedObjects.getScopedClient(context.request, {
+      includedHiddenTypes: [SEARCH_SESSION_TYPE],
+    });
+    const finder = savedObjectsClient.createPointInTimeFinder<SearchSessionAttributes>({
+      type: 'search-session',
+      perPage: 1000,
+      fields: ['name', 'username', 'expires'],
+      namespaces: ['*'],
+    });
+
+    const searchSessions: Array<SavedObjectsFindResult<SearchSessionAttributes>> = [];
+
+    for await (const response of finder.find()) {
+      searchSessions.push(
+        ...response.saved_objects.filter(
+          (so) => new Date(so.attributes.expires).getTime() > Date.now()
+        )
+      );
+    }
+
+    if (!searchSessions.length) {
+      return [];
+    }
+
+    return [
+      {
+        title: i18n.translate('data.deprecations.searchSessionsTitle', {
+          defaultMessage: 'Found active search sessions',
+        }),
+        message: buildMessage({ searchSessions }),
+        deprecationType: 'feature',
+        level: 'warning',
+        correctiveActions: {
+          manualSteps: [
+            i18n.translate('data.deprecations.searchSessions.manualStepOneMessage', {
+              defaultMessage: 'Open the kibana.yml config file.',
+            }),
+            i18n.translate('data.deprecations.searchSessions.manualStepTwoMessage', {
+              defaultMessage: 'Add the following: "data.search.sessions.enabled: true"',
```

Comment:
There is another warning if the setting is added:

<img width="1302" alt="Screenshot 2025-01-21 at 18 17 27" src="https://github.com/user-attachments/assets/49b26eeb-a7e6-4167-ae59-755ca5c7f63b" />

Should we remove or update this warning now? Otherwise it's confusing that the new instructions are to add the setting which is not supposed to be available in 9.0 according to this flyout message "will be removed in 9".

Reply:
Great catch, I need to remove this warning now that it's sticking around for 9.0